app/src/main/java/org/wikipedia/feed/view/GraphView.kt

```diff
@@ -0,0 +1,61 @@
+package org.wikipedia.feed.view
+
+import android.content.Context
+import android.graphics.*
+import android.util.AttributeSet
+import android.view.View
+import androidx.core.content.ContextCompat
+import org.wikipedia.R
+import org.wikipedia.feed.mostread.MostReadArticles
+
+class GraphView(context: Context, attributeSet: AttributeSet) : View(context, attributeSet) {
+    private val gradientColor1 = ContextCompat.getColor(context, R.color.accent50)
+    private val gradientColor2 = ContextCompat.getColor(context, R.color.green50)
+    private val dataSet = mutableListOf<MostReadArticles.ViewHistory>()
+    private var maxX = 0
+    private var maxY = 0
+    private val path = Path()
+
+    private val pathPaint = Paint().apply {
+        style = Paint.Style.STROKE
+        strokeWidth = GRAPH_STROKE_WIDTH
+        strokeCap = Paint.Cap.ROUND
+        pathEffect = CornerPathEffect(GRAPH_STROKE_WIDTH)
+        isAntiAlias = true
+    }
+
+    fun setData(viewHistoryList: List<MostReadArticles.ViewHistory>) {
```

Comment:
It's nice that this class has the generic name `GraphView`, which suggests we'll be able to reuse it for graphing other things.  But then it's using this extremely specific `MostReadArticles.ViewHistory` object for providing the data points. Is there a way for this View class to take a completely generic list of `floats` for graphing, regardless of where the numbers come from?

Reply:
Make sense. Done!